engine-extensions/src/main/java/io/nosqlbench/engine/extensions/computefunctions/ComputeFunctions.java

```diff
@@ -146,14 +156,66 @@ public static int[] intersection(int[] reference, int[] sample, int limit) {
     public static int intersectionSize(int[] reference, int[] sample) {
         return Intersections.count(reference, sample);
     }
+
     public static int intersectionSize(int[] reference, int[] sample, int limit) {
         return Intersections.count(reference, sample, limit);
     }
+
     public static int intersectionSize(long[] reference, long[] sample) {
         return Intersections.count(reference, sample);
     }
+
     public static int intersectionSize(long[] reference, long[] sample, int limit) {
         return Intersections.count(reference, sample, limit);
     }
 
+    public static double F1(int[] reference, int[] sample) {
+        return F1(reference, sample, reference.length);
+    }
+
+    public static double F1(int[] reference, int[] sample, int limit) {
+        double recallAtK = recall(reference, sample, limit);
+        double precisionAtK = precision(reference, sample, limit);
```

Comment:
Is there a way to reuse prior metric results from the same observation? Or does that introduce complexity that would eliminate any perceived performance gain?